drivers: spi: litex: soc.h not included, but needed

**Describe the bug**
`soc.h` is not included in `spi_litespi.c`

**To Reproduce**
execute test: `tests/drivers/spi/dt_spec/drivers.spi.dt_spec` for board `litex_vexriscv`

**Expected behavior**
`soc.h` should be included in `spi_litespi.c`

**Impact**
What impact does this issue have on your progress (e.g., annoyance, showstopper)
CI builds for `litex_vexriscv` fail

**Logs and console output**
```
/__w/zephyr/zephyr/drivers/spi/spi_litespi.c: In function 'spi_config':
/__w/zephyr/zephyr/drivers/spi/spi_litespi.c:71:17: error: implicit declaration of function 'litex_write8' [-Werror=implicit-function-declaration]
   71 |                 litex_write8(SPI_ENABLE, SPI_LOOPBACK_ADDR);
      |                 ^~~~~~~~~~~~
/__w/zephyr/zephyr/drivers/spi/spi_litespi.c:78:9: error: implicit declaration of function 'litex_write16'; did you mean 'sys_write16'? [-Werror=implicit-function-declaration]
   78 |         litex_write16(*control, SPI_CONTROL_ADDR);
      |         ^~~~~~~~~~~~~
      |         sys_write16
/__w/zephyr/zephyr/drivers/spi/spi_litespi.c: In function 'spi_litespi_send':
/__w/zephyr/zephyr/drivers/spi/spi_litespi.c:91:18: error: implicit declaration of function 'litex_read8' [-Werror=implicit-function-declaration]
   91 |         while (!(litex_read8(SPI_STATUS_ADDR)))
      |                  ^~~~~~~~~~~
```

**Environment (please complete the following information):**
NA

**Additional context**
NA

--- a/drivers/spi/spi_litespi.c
+++ b/drivers/spi/spi_litespi.c
@@ -11,6 +11,7 @@
 LOG_MODULE_REGISTER(spi_litespi);
 #include "spi_litespi.h"
 #include <stdbool.h>
+#include <soc.h>
 
 /* Helper Functions */
 static int spi_config(const struct spi_config *config, uint16_t *control)

